extensions/amp-carousel/0.1/base-carousel.js

```diff
@@ -25,17 +25,66 @@ export class BaseCarousel extends AMP.BaseElement {
     /** @private {!Element} */
     this.nextButton_;
 
-    this.buildCarousel();
-    this.buildButtons();
-    this.setupGestures();
-    this.setControlsState();
+    /** @private {?number} */
+    this.autoplayTimeoutId_ = null;
```

Comment:
I am fine with moving it to another base class. 

But here is why this would work here , 

scroller has functionality to move next and prev (achieved via `go`)
And autoplay uses `go` to do the autoplay.
It is not illogical to have autoplay on a scrollable page with left and right navigation.